### PR TITLE
Update ScannerDB image

### PR DESCRIPTION
**Pull request by RTann**
Install Postgres 12 via instructions inspired by https://www.postgresql.org/download/linux/redhat/.
This removes the need to install individual rpms.

Updated the docker-entrypoint by copying https://github.com/docker-library/postgres/blob/master/12/bullseye/docker-entrypoint.sh

Added `STACKROX MODIFIED` to any comments added by this repo

### CODE REVIEW

File: image/db/rhel/scripts/docker-entrypoint.sh [hunk 1]
```diff
@@ -33,19 +33,20 @@ _is_sourced() {
...
 
-# used to create initial posgres directories and if run as root, ensure ownership to the "postgres" user
+# used to create initial postgres directories and if run as root, ensure ownership to the "postgres" user
 docker_create_db_directories() {
 	local user; user="$(id -u)"
 
 	mkdir -p "$PGDATA"
-	chmod 700 "$PGDATA"
+	# ignore failure since there are cases where we can't chmod (and PostgreSQL might fail later anyhow - it's picky about permissions of this directory)
+	chmod 700 "$PGDATA" || :
```
Comment (c-du) [inline image/db/rhel/scripts/docker-entrypoint.sh:42]: Did you see real error for this?

File: image/db/rhel/Dockerfile [hunk 2]
```diff
@@ -22,24 +21,21 @@ ENV PATH="$PATH:/usr/pgsql-$PG_MAJOR/bin/" \
...
 RUN groupadd -g 70 postgres && \
     adduser postgres -u 70 -g 70 -d /var/lib/postgresql -s /bin/sh && \
-    rpm --import RPM-GPG-KEY-PGDG-12 && \
     dnf upgrade -y && \
     dnf install -y \
-        ca-certificates libicu systemd-sysv /tmp/postgres-libs.rpm \
-        /tmp/postgres-server.rpm /tmp/postgres.rpm glibc-locale-source glibc-langpack-en && \
+        ca-certificates libicu systemd-sysv glibc-locale-source glibc-langpack-en \
+        https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm && \
+    dnf install -y postgresql12-server && \
```
Comment (c-du) [inline image/db/rhel/Dockerfile:33]: Good way to go.

File: image/db/rhel/scripts/docker-entrypoint.sh (hunk 1 above)
Comment (RTann) [reply image/db/rhel/scripts/docker-entrypoint.sh:42]: No, this is just directly lifted from https://github.com/docker-library/postgres/blob/master/12/bullseye/docker-entrypoint.sh

We haven't updated this file in years, so I just copy pasted the latest version here and added comments in sections marked with `STACKROX MODIFIED`

**Review by c-du — APPROVED** [review]

**Comment by RTann** [comment]
Going to merge this after the next Scanner release is cut (which will be soon), so we have plenty of time for this to bake for the next release. Once merged I will make an equivalent PR in the stackrox repo

**Pull request merged**